storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,28 +1,276 @@
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(AzureQueueStorageImpl client, String queueName) {
+        this.queueName = queueName;
+        this.client = new AzureQueueStorageImpl(client.httpPipeline())
+            .withUrl(client.url())
+            .withVersion(client.version());
+    }
+
+    /**
+     * Constructor used by the builder
+     * @param endpoint URL of the queue
+     * @param httpPipeline Http pipeline
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(URL endpoint, HttpPipeline httpPipeline, String queueName) {
+        this.queueName = queueName;
+        this.client = new AzureQueueStorageImpl(httpPipeline)
+            .withUrl(endpoint.toString());
     }
 
     /**
-     * Creates a builder that can configure options for the SecretAsyncClient before creating an instance of it.
-     * @return A new builder to create a SecretAsyncClient from.
+     * @return a new client builder instance
      */
     public static QueueAsyncClientBuilder builder() {
         return new QueueAsyncClientBuilder();
     }
+
+    /**
+     * @return the URL of the queue
+     */
+    public String url() {
+        return client.url();
+    }
+
+    /**
+     * Creates a queue with no metadata
+     * @return an empty response
+     */
+    public Mono<VoidResponse> create() {
+        return create(null);
+    }
+
+    /**
+     * Creates a queue with metadata
+     * @param metadata Metadata to set on the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> create(Map<String, String> metadata) {
+        return client.queues().createWithRestResponseAsync(queueName, null, metadata, null, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Deletes the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> delete() {
+        return client.queues().deleteWithRestResponseAsync(queueName, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * @return the properties of the queue
+     */
+    public Mono<Response<QueueProperties>> getProperties() {
+        return client.queues().getPropertiesWithRestResponseAsync(queueName, Context.NONE)
+            .map(this::getQueuePropertiesResponse);
+    }
+
+    /**
+     * Sets the metadata of the queue
+     * @param metadata Metadata to set on the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> setMetadata(Map<String, String> metadata) {
+        return client.queues().setMetadataWithRestResponseAsync(queueName, null, metadata, null, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * @return the access policies of the queue
+     */
+    public Flux<SignedIdentifier> getAccessPolicy() {
+        return client.queues().getAccessPolicyWithRestResponseAsync(queueName, Context.NONE)
+            .flatMapMany(response -> Flux.fromIterable(response.value()));
```

Comment:
This seems weird - you're converting from an Iterable to a Flux? That seems back to front to me. Especially so if it means you're pulling down the whole iterable content first.

Reply:
It's true. The API returns a Mono`<Response<List<SignedIdentifier>>>`.